2025-09-18 11:12:29,144 : INFO : root : Loading config.
2025-09-18 11:12:29,148 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2025-09-18 11:12:29,148 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\cfa6ecba65704a931c9dd214c44986c8.mpf_cache
2025-09-18 11:12:29,149 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\White\Documents\Top_Gun_Pinball\config\config_lights.yaml
2025-09-18 11:12:29,150 : INFO : ConfigProcessor : Loading config from file C:\Users\White\mpf\Lib\site-packages\mpf\mpfconfig.yaml.
2025-09-18 11:12:29,164 : INFO : ConfigProcessor : Loading config: C:\Users\White\mpf\Lib\site-packages\mpf\mpfconfig.yaml
2025-09-18 11:12:29,164 : INFO : ConfigProcessor : Loading config from file C:\Users\White\Documents\Top_Gun_Pinball\config\config.yaml.
2025-09-18 11:12:29,166 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config.yaml
2025-09-18 11:12:29,169 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_autofire_coils.yaml
2025-09-18 11:12:29,171 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_ball_devices.yaml
2025-09-18 11:12:29,178 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_coils.yaml
2025-09-18 11:12:29,180 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_flippers.yaml
2025-09-18 11:12:29,183 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_hardware_fast.yaml
2025-09-18 11:12:29,205 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_lights.yaml
2025-09-18 11:12:29,213 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_switches.yaml
2025-09-18 11:12:29,215 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\variable_player.yaml
2025-09-18 11:12:29,222 : INFO : ConfigProcessor : Config file cache created: C:\Users\White\AppData\Local\Temp\cfa6ecba65704a931c9dd214c44986c8.mpf_cache
2025-09-18 11:12:29,334 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f98ffc0ff5962effface7d86bc406d5c.mpf_cache
2025-09-18 11:12:29,335 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\91cc4d1c993c58dc354e34974b9cfeac.mpf_cache
2025-09-18 11:12:29,336 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f98ffc0ff5962effface7d86bc406d5c.mpf_cache
2025-09-18 11:12:29,336 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\76792f1d686f89709887dba7bd69667a.mpf_cache
2025-09-18 11:12:29,337 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\142bbae4707335cc8e3efaa4364c4893.mpf_cache
2025-09-18 11:12:29,337 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\b4249e7b8f95e82c549b9263ef475a09.mpf_cache
2025-09-18 11:12:29,338 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\feb97b01eed59f715237372ed3ab4fda.mpf_cache
2025-09-18 11:12:29,339 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\434e0238e97461d536537fa1d2ac13b8.mpf_cache
2025-09-18 11:12:29,340 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\5cb8a848aaedecad105d66ca86abc153.mpf_cache
2025-09-18 11:12:29,340 : INFO : Machine : Mission Pinball Framework Core Engine v0.80.0.dev9
2025-09-18 11:12:29,340 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'platform_integration_test': False, 'jsonlogging': False, 'logfile': 'logs\\2025-09-18-11-12-29-mpf-TopGunPinball.log', 'pause': False, 'production': False, 'text_ui': False, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'virtual', 'syslog_address': None, 'full_logfile_path': 'C:\\Users\\White\\Documents\\Top_Gun_Pinball\\logs\\2025-09-18-11-12-29-mpf-TopGunPinball.log'}
2025-09-18 11:12:29,340 : INFO : Machine : MPF path: C:\Users\White\mpf\Lib\site-packages\mpf
2025-09-18 11:12:29,340 : INFO : Machine : Machine path: C:\Users\White\Documents\Top_Gun_Pinball
2025-09-18 11:12:29,340 : INFO : Machine : Platform: win32
2025-09-18 11:12:29,340 : INFO : Machine : Python executable location: C:\Users\White\mpf\Scripts\python.exe
2025-09-18 11:12:29,340 : INFO : Machine : Python version: 3.12.7 (64-bit)
2025-09-18 11:12:29,343 : INFO : Machine : Initializing MPF...
2025-09-18 11:12:29,441 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,469 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2025-09-18 11:12:29,469 : INFO : EventManager : Event: ======'machine_var_player1_score'====== Args={'value': 61000, 'prev_value': None, 'change': True}
2025-09-18 11:12:29,469 : INFO : EventManager : Event: ======'machine_var_player2_score'====== Args={'value': 73000, 'prev_value': None, 'change': True}
2025-09-18 11:12:29,469 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev9', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,469 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev9, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,470 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.12.7', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-11-10.0.26100-SP0', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '11', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.26100', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'machine_var_log_file_path'====== Args={'value': 'C:\\Users\\White\\Documents\\Top_Gun_Pinball\\logs\\2025-09-18-11-12-29-mpf-TopGunPinball.log', 'prev_value': None, 'change': True}
2025-09-18 11:12:29,624 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2025-09-18 11:12:29,672 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2025-09-18 11:12:29,673 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-0', <Platform.Virtual>) <switch.s_trough_1>
2025-09-18 11:12:29,673 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-1', <Platform.Virtual>) <switch.s_trough_2>
2025-09-18 11:12:29,673 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-2', <Platform.Virtual>) <switch.s_trough_3>
2025-09-18 11:12:29,673 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-3', <Platform.Virtual>) <switch.s_trough_4>
2025-09-18 11:12:29,673 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-4', <Platform.Virtual>) <switch.s_trough_5>
2025-09-18 11:12:29,673 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-5', <Platform.Virtual>) <switch.s_trough_6>
2025-09-18 11:12:29,675 : INFO : EventManager : Event: ======'bcp_connection_attempt'====== Args={'name': 'local_display', 'host': 'localhost', 'port': 5050}
2025-09-18 11:12:29,677 : INFO : BCPClientSocket : Connecting BCP to 'local_display' at localhost:5050...
2025-09-18 11:12:29,689 : INFO : BCPClientSocket : Connected BCP to 'local_display' localhost:5050
2025-09-18 11:12:29,690 : INFO : EventManager : Event: ======'bcp_clients_connected'====== Args={}
2025-09-18 11:12:29,690 : INFO : ball_device.bd_plunger : BCH: Initial count: 0
2025-09-18 11:12:29,690 : INFO : ball_device.bd_plunger : State: idle
2025-09-18 11:12:29,691 : INFO : ball_device.bd_trough : BCH: Initial count: 6
2025-09-18 11:12:29,691 : INFO : ball_device.bd_trough : State: idle
2025-09-18 11:12:29,691 : INFO : ball_device.bd_lock_elevator : BCH: Initial count: 0
2025-09-18 11:12:29,691 : INFO : ball_device.bd_lock_elevator : State: idle
2025-09-18 11:12:29,691 : INFO : ball_device.bd_mode_scoop : BCH: Initial count: 0
2025-09-18 11:12:29,692 : INFO : ball_device.bd_mode_scoop : State: idle
2025-09-18 11:12:29,707 : INFO : EventManager : Event: ======'init_phase_3'====== Args={}
2025-09-18 11:12:29,710 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...e_o_left\']')>)
handle: <Handle EventManager._queue_task_done(<Task finishe...e_o_left\']')>)>
Traceback (most recent call last):
  File "C:\Users\White\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\show_controller.py", line 53, in _load_shows
    self.machine.shows[show_name].load(show_config)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\assets\show.py", line 171, in load
    self._process_step_actions(step, actions)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\assets\show.py", line 201, in _process_step_actions
    raise e
  File "C:\Users\White\mpf\Lib\site-packages\mpf\assets\show.py", line 197, in _process_step_actions
    self.machine.show_controller.show_players[key].validate_config_entry(value, self.name)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 85, in _validate_config_item
    devices = self._expand_device(device)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 114, in _expand_device
    return self.raise_config_error(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\logging.py", line 175, in raise_config_error
    raise ConfigFileError(msg, error_no, self.log.name if self.log else "", context, self._url_base) \
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in light_player: Show "attract_circle_all" >> Failed to load config player attract_circle_all:lights with settings {'l_gi_light_9': 'off', 'l_gi_light_3': 'off', 'l_gi_light_4': 'off', 'l_gi_light_5': 'off', 'l_gi_light_6': 'off', 'l_gi_light_11': 'off', 'l_gi_light_12': 'off', 'l_gi_light_16': 'off', 'l_gi_light_19': 'off', 'l_gi_light_20': 'off', 'l_gi_light_21': 'off', 'l_gi_light_22': 'off', 'l_gi_light_23': 'off', 'l_gi_light_17': 'off', 'l_gi_light_18': 'off', 'l_gi_light_7': 'off', 'l_gi_light_10': 'off', 'l_gi_light_1': 'off', 'l_gi_light_2': 'off', 'l_gi_light_15': 'off', 'l_gi_light_14': 'off', 'l_gi_light_13': 'off', 'l_eject': 'off', 'l_shoot_again': 'off', 'l_shot_left_lane': 'off', 'l_shot_left_orbit': 'off', 'l_goose_g': 'off', 'l_gooose_o_left': 'off', 'l_iceman_i': 'off', 'l_iceman_c': 'off', 'l_iceman_e': 'off', 'l_shot_left_ramp': 'off', 'l_shot_spinner': 'off', 'l_shot_barrier': 'off', 'l_shot_middle_ramp': 'off', 'l_shot_upper_orbit': 'off', 'l_shot_tower_ramp': 'off', 'l_lock_c': 'off', 'l_lock_mav': 'off', 'l_lock_goose': 'off', 'l_shot_right_orbit': 'off', 'l_goose_o_right': 'off', 'l_goose_s': 'off', 'l_goose_e': 'off', 'l_iceman_m': 'off', 'l_iceman_a': 'off', 'l_iceman_n': 'off', 'l_lock_k': 'off', 'l_shot_pilot_gear': 'off', 'l_lock_l': 'off', 'l_lock_o': 'off', 'l_mig_m': 'off', 'l_mig_i': 'off', 'l_mig_g': 'off'} >> Could not find a lights device with name or tag l_gooose_o_left, from list ['l_gooose_o_left'] Error Code: CFE-light_player-101 (https://missionpinball.org/logs)
2025-09-18 11:12:29,827 : INFO : Machine : Starting the main run loop with active modes: []
2025-09-18 11:12:29,827 : WARNING : Machine : Attract mode is not active, game will not be playable. Please check your attract mode configuration.
2025-09-18 11:12:29,828 : INFO : Machine : Shutting down...
2025-09-18 11:12:29,828 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2025-09-18 11:12:30,830 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\machine.py", line 808, in _run_loop
    raise self._exception['exception']
  File "C:\Users\White\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\show_controller.py", line 53, in _load_shows
    self.machine.shows[show_name].load(show_config)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\assets\show.py", line 171, in load
    self._process_step_actions(step, actions)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\assets\show.py", line 201, in _process_step_actions
    raise e
  File "C:\Users\White\mpf\Lib\site-packages\mpf\assets\show.py", line 197, in _process_step_actions
    self.machine.show_controller.show_players[key].validate_config_entry(value, self.name)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 85, in _validate_config_item
    devices = self._expand_device(device)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 114, in _expand_device
    return self.raise_config_error(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\logging.py", line 175, in raise_config_error
    raise ConfigFileError(msg, error_no, self.log.name if self.log else "", context, self._url_base) \
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in light_player: Show "attract_circle_all" >> Failed to load config player attract_circle_all:lights with settings {'l_gi_light_9': 'off', 'l_gi_light_3': 'off', 'l_gi_light_4': 'off', 'l_gi_light_5': 'off', 'l_gi_light_6': 'off', 'l_gi_light_11': 'off', 'l_gi_light_12': 'off', 'l_gi_light_16': 'off', 'l_gi_light_19': 'off', 'l_gi_light_20': 'off', 'l_gi_light_21': 'off', 'l_gi_light_22': 'off', 'l_gi_light_23': 'off', 'l_gi_light_17': 'off', 'l_gi_light_18': 'off', 'l_gi_light_7': 'off', 'l_gi_light_10': 'off', 'l_gi_light_1': 'off', 'l_gi_light_2': 'off', 'l_gi_light_15': 'off', 'l_gi_light_14': 'off', 'l_gi_light_13': 'off', 'l_eject': 'off', 'l_shoot_again': 'off', 'l_shot_left_lane': 'off', 'l_shot_left_orbit': 'off', 'l_goose_g': 'off', 'l_gooose_o_left': 'off', 'l_iceman_i': 'off', 'l_iceman_c': 'off', 'l_iceman_e': 'off', 'l_shot_left_ramp': 'off', 'l_shot_spinner': 'off', 'l_shot_barrier': 'off', 'l_shot_middle_ramp': 'off', 'l_shot_upper_orbit': 'off', 'l_shot_tower_ramp': 'off', 'l_lock_c': 'off', 'l_lock_mav': 'off', 'l_lock_goose': 'off', 'l_shot_right_orbit': 'off', 'l_goose_o_right': 'off', 'l_goose_s': 'off', 'l_goose_e': 'off', 'l_iceman_m': 'off', 'l_iceman_a': 'off', 'l_iceman_n': 'off', 'l_lock_k': 'off', 'l_shot_pilot_gear': 'off', 'l_lock_l': 'off', 'l_lock_o': 'off', 'l_mig_m': 'off', 'l_mig_i': 'off', 'l_mig_g': 'off'} >> Could not find a lights device with name or tag l_gooose_o_left, from list ['l_gooose_o_left'] Error Code: CFE-light_player-101 (https://missionpinball.org/logs)
2025-09-18 11:12:30,849 : INFO : root : MPF run loop ended.
